### PR TITLE
✨ Spaces persistent storage

### PR DESCRIPTION
**Pull request by SBrandeis**
Related PR (internal link): https://github.com/huggingface/moon-landing/pull/6530

### CODE REVIEW

File: docs/hub/spaces-storage.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+You can upgrade your Space to have access to a persistent disk space from the **Settings** tab.
+
+
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/spaces-storage-settings.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/spaces-storage-settings-dark.png"/>
+</div>
+
+By upgrading your Space, you get access to a some disk space that persists across restarts of your Space.
```
Comment (christophe-rannou) [inline docs/hub/spaces-storage.md:18]:
```suggestion
By upgrading your Space, you get access to some disk space that persists across restarts of your Space.
```

File: docs/hub/spaces-storage.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+|------------------	|------------------	|------------------	|----------------------	|
+| Free tier        	| 50GB            	| No (ephemeral)  	| Free!                	|
+| Small           	| 20GB             	| Yes           	| $5                	|
+| Medium         	| 150 GB           	| Yes           	| $25                	|
+| Large         	| 1TB              	| Yes           	| $100                	|
+
+
+### Billing
+
+Billing on Spaces is based on hardware usage and is computed by the minute: you get charged for every minute the Space runs on the requested hardware, regardless of whether the Space is used.
```
Comment (Michellehbn) [inline docs/hub/spaces-storage.md:44]:
```suggestion
Billing of Spaces is based on hardware usage and is computed by the minute: you get charged for every minute the Space runs on the requested hardware, regardless of whether the Space is used.
```

File: docs/hub/spaces-storage.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+Because it requires resources even when the Space is not running, persistent storage upgrades are billed for all their lifetime until deleted, regardless of the Space status and running state.
+
+Additional information about billing can be found in the [dedicated Hub-wide section](./billing).
+
+## Dataset storage
+
+If you do not want to pay for a persistent disk storage, but still need to persist data that lives longer than your Space, you can
```
Comment (Michellehbn) [inline docs/hub/spaces-storage.md:52]:
```suggestion
If you need to persist data that lives longer than your Space, you could 
```

**Review by Michellehbn — COMMENTED** [review]
🚀

File: docs/hub/spaces-storage.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+That means you can `read` and `write to` this storage from your Space as you would with a traditional hard-drive or SSD.
+
+Persistent disk Space can be upgraded to a larger tier at will, though cannot be downgraded to a smaller tier. If you wish to use a smaller persistent storage tier, you must delete your current (larger) storage first.
+
+<Tip warning={true}>
+	WARNING: all data stored in the storage is lost when you delete it.
+</Tip>
+
+### Persitent storage specs
```
Comment (Michellehbn) [inline docs/hub/spaces-storage.md:30]:
```suggestion
### Persistent storage specs
```

File: docs/hub/spaces-storage.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+Here are the specifications for each of the different upgrade options:
+
+| **Tier**        	| **Disk space** 	| **Persistent** 	| **Monthly Price** 	|
```
Comment (Michellehbn) [inline docs/hub/spaces-storage.md:34]:
```suggestion
| **Tier**        	| **Disk Space** 	| **Persistent** 	| **Monthly Price** 	|
```

**Review by Michellehbn — COMMENTED** [review]
Quick typo fix ❤️

File: docs/hub/spaces-storage.md [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/spaces-storage-settings.png"/>
+<img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/spaces-storage-settings-dark.png"/>
+</div>
+
+You can choose the storage tier of your choice to get access to disk space that persists across restarts of your Space.
+
+Persistent storage acts like a traditional disk storage, mounted on `/data`.
+
+That means you can `read` and `write to` this storage from your Space as you would with a traditional hard-drive or SSD.
```
Comment (osanseviero) [inline docs/hub/spaces-storage.md:22]:
```suggestion
That means you can `read` and `write to` this storage from your Space as you would with a traditional hard drive or SSD.
```

File: docs/hub/spaces-storage.md [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+Persistent disk Space can be upgraded to a larger tier at will, though cannot be downgraded to a smaller tier. If you wish to use a smaller persistent storage tier, you must delete your current (larger) storage first.
```
Comment (osanseviero) [inline docs/hub/spaces-storage.md:24]:
```suggestion
Persistent disk space can be upgraded to a larger tier at will, though it cannot be downgraded to a smaller tier. If you wish to use a smaller persistent storage tier, you must delete your current (larger) storage first.
```

**Review by osanseviero — COMMENTED** [review]
Very nice! As discussed in Slack

* We can also document how to use this in Docker, likely in the Docker section
* I would add some content in https://huggingface.co/docs/hub/spaces-overview#hardware-resources
* As discussed with @hysts, we can document how to make the libraries store to `/data` for Space development

File: docs/hub/spaces-storage.md (hunk 7 above)
Comment (Michellehbn) [reply docs/hub/spaces-storage.md:24]: There's no need to add "it" (in the second clause before cannot) bc it is assumed grammatically but we can add it that's fine with me 🤗

**Review by osanseviero — APPROVED** [review]
Thanks!

**Review by julien-c — APPROVED** [review]

**Pull request merged**